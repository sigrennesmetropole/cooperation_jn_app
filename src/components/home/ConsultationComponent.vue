<script setup lang="ts">
import fabriqueCitoyenne from '@/assets/icons/fabrique-citoyenne.svg'
import router from '@/router'
import { usePanelsStore } from '@/stores/panels'
import { computed } from 'vue'

const panelStore = usePanelsStore()

const leftAlignment = computed(() =>
  panelStore.isInformationPanelShown
    ? 'absolute left-[462px] top-3'
    : 'absolute left-3 top-3'
)

function goToConsultation() {
  router.push('/consultation')
}
</script>

<template>
  <div
    class="z-10 flex flex-row gap-4 p-2 bg-[#261E69] rounded-lg text-white items-center"
    :class="leftAlignment"
  >
    <div class="flex flex-row gap-3 px-2">
      <img :src="fabriqueCitoyenne" class="w-9 h-9 my-auto" />
      <div>
        <h4 class="font-dm-sans font-bold text-lg leading-6">
          La Fabrique Citoyenne
        </h4>
        <p class="font-dm-sans font-normal text-sm">
          Découvrez l'espace concertation et <br />
          contribuez au transport de demain
        </p>
      </div>
    </div>
    <button
      tabindex="0"
      class="font-dm-sans font-medium text-base border border-white px-4 py-3 rounded-lg h-12"
      @click="goToConsultation()"
      @keydown.enter="goToConsultation()"
    >
      Découvrir
    </button>
  </div>
</template>
