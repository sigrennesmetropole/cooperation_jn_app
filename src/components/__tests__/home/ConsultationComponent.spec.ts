import ConsultationComponent from '@/components/home/ConsultationComponent.vue'
import { describe, it, expect, beforeEach, vi } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'

describe('ConsultationComponent.vue', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(ConsultationComponent, {
      global: {
        plugins: [
          createTestingPinia({
            createSpy: vi.fn,
            stubActions: false,
            stubPatch: false,
            fakeApp: true,
          }),
        ],
      },
    })
  })

  it('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
