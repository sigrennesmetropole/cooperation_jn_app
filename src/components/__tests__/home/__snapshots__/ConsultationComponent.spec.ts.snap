// Vitest Snapshot v1

exports[`ConsultationComponent.vue > renders the component correctly 1`] = `
"<div class=\\"z-10 flex flex-row gap-4 p-2 bg-[#261E69] rounded-lg text-white items-center absolute left-[462px] top-3\\">
  <div class=\\"flex flex-row gap-3 px-2\\"><img src=\\"/src/assets/icons/fabrique-citoyenne.svg\\" class=\\"w-9 h-9 my-auto\\">
    <div>
      <h4 class=\\"font-dm-sans font-bold text-lg leading-6\\"> La Fabrique Citoyenne </h4>
      <p class=\\"font-dm-sans font-normal text-sm\\"> Découvrez l'espace concertation et <br> contribuez au transport de demain </p>
    </div>
  </div><button tabindex=\\"0\\" class=\\"font-dm-sans font-medium text-base border border-white px-4 py-3 rounded-lg h-12\\"> Découvrir </button>
</div>"
`;
